### PR TITLE
logic-screen

### PR DESCRIPTION
**Pull request by Pratham3006**
PR for login-screen

### CODE REVIEW

File: App.tsx [hunk 1]
```diff
@@ -1,8 +1,11 @@
...
 // NAVIGATION //
-import AppNavigation from "./src/infrastructure/navigation/AuthStack";
-import { NavigationContainer } from "@react-navigation/native";
+import AppNavigation from "./src/infrastructure/navigation/AppNavigation";
+
+// PROVIDERS //
+import { UserProvider } from "./src/contexts/user.context";
+import { AuthenticationProvider } from "./src/contexts/authentication.context";
 
 // PROVIDERS //
```
Comment (anchanadarsh) [inline App.tsx]: remove this

File: src/contexts/authentication.context.tsx [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+		setUser(null);
+		flushLocalStorage();
+		setIsAuthLoading(false);
+	};
+
+	/** Logout the User */
+	const logout = async (): Promise<void> => {
+		// Start changing Auth State
+		setIsAuthLoading(true);
+		const userId = (await getDataFromLocalStorage(LocalStorageKeys.USER))._id;
```
Comment (anchanadarsh) [inline src/contexts/authentication.context.tsx:90]: Add comment 
// Get the userId from Local storage

File: src/contexts/authentication.context.tsx [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+			console.log(error);
+			setIsAuthLoading(false);
+			return {
+				status: false,
+				message: "Error Logging In",
+			};
+		}
+	};
+
+	/** Login the Usr Locally (within the App) */
```
Comment (anchanadarsh) [inline src/contexts/authentication.context.tsx]: Usr -> User

File: src/contexts/authentication.context.tsx [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+	const loginLocally = async (user: UserData, token: string): Promise<void> => {
+		// Save the User and Token to Local Storage
+		await setDataInLocalStorage(LocalStorageKeys.USER, user);
+		await setDataInLocalStorage(LocalStorageKeys.TOKEN, token);
+		setUser(user);
```
Comment (anchanadarsh) [inline src/contexts/authentication.context.tsx:152]: Add comments
// Save user data in the user context

File: src/contexts/authentication.context.tsx [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+		setIsAuthenticated(true);
+		setIsAuthLoading(false);
+	};
+
+	/** Check if the user is logged in */
+	const checkLoggedIn = async (): Promise<void> => {
+		try {
+			// Start the Verification Pro
+			setIsAuthLoading(true);
+			const userObject = await getDataFromLocalStorage(LocalStorageKeys.USER);
```
Comment (anchanadarsh) [inline src/contexts/authentication.context.tsx:166]: Add comments
// Get user and token from Local storage

File: src/infrastructure/navigation/AppNavigation.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+// UTILS //
+import { CONSTANTS } from "../constants";
+
+// NAVIGATION //
+import { NavigationContainer } from "@react-navigation/native";
+import { createStackNavigator } from "@react-navigation/stack";
+
+// OTHERS //
+import AuthStack from "./AuthStack";
+import HomeScreen from "../../features/home/screens/Home.screen";
```
Comment (anchanadarsh) [inline src/infrastructure/navigation/AppNavigation.tsx]: segregate imports

File: src/infrastructure/navigation/AppNavigation.tsx [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+		<NavigationContainer>
+			{/* If Android then put a Custom Status Bar */}
+			{CONSTANTS.IS_ANDROID && (
+				<StatusBar
+					barStyle="light-content"
+					backgroundColor="transparent"
+					translucent
+				/>
+			)}
+			{!isAuthenticated ? (
```
Comment (anchanadarsh) [inline src/infrastructure/navigation/AppNavigation.tsx:42]: Add comments
If user is authenticated then show AuthStack else show HomeScreen

**Review by anchanadarsh — CHANGES_REQUESTED** [review]

File: src/contexts/authentication.context.tsx [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+} from "../services/api/users.api-service";
+
+// SERVICES //
+import {
+	flushLocalStorage,
+	getDataFromLocalStorage,
+	setDataInLocalStorage,
+} from "../services/local-storage.service";
+
+// OTHERS //
```
Comment (aminabhay) [inline src/contexts/authentication.context.tsx]: CONTEXTS

File: src/contexts/authentication.context.tsx [hunk 9]
```diff
@@ -0,0 +1,218 @@
...
+		}
+	};
+
+	/** Login the User Locally (within the App) */
+	const loginLocally = async (user: UserData, token: string): Promise<void> => {
+		// Save the User and Token to Local Storage
+		await setDataInLocalStorage(LocalStorageKeys.USER, user);
+		await setDataInLocalStorage(LocalStorageKeys.TOKEN, token);
+
+		// Save user data in the user context
```
Comment (aminabhay) [inline src/contexts/authentication.context.tsx]: Capital U

File: src/contexts/authentication.context.tsx [hunk 10]
```diff
@@ -0,0 +1,218 @@
...
+		setIsAuthLoading(false);
+	};
+
+	/** Check if the user is logged in */
+	const checkLoggedIn = async (): Promise<void> => {
+		try {
+			// Start the Verification Pro
+			setIsAuthLoading(true);
+
+			// Get user and token from Local storage
```
Comment (aminabhay) [inline src/contexts/authentication.context.tsx]: capital U

File: src/enums/analytics.ts [hunk 11]
```diff
@@ -0,0 +1,4 @@
+export enum FirebaseAnalyticsEvents {
```
Comment (aminabhay) [inline src/enums/analytics.ts]: just call it AnalyticsEvents

File: src/features/account/screens/Login.screen.tsx [hunk 12]
```diff
@@ -0,0 +1,170 @@
...
+					...prevErrors,
+					form_error: "Looks like we faced some network issues. Please try again.",
+				}));
+			}
+		}
+	};
+
+	// Use Effect and Focus Effect
+	useEffect(() => {
+		// Log Event in firebase analytics when Login Screen is opened
```
Comment (aminabhay) [inline src/features/account/screens/Login.screen.tsx]: Analytics
Don't use the word Firebase, keep it platform agnostic

File: src/infrastructure/constants.ts [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+// Define constant values
+export const CONSTANTS = {
+	WINDOW_HEIGHT: Dimensions.get("window").height,
+	WINDOW_WIDTH: Dimensions.get("window").width,
+	OS: Platform.OS,
+	IS_ANDROID: Platform.OS === "android",
+	IS_IOS: Platform.OS === "ios",
+	VERSION: Platform.Version,
+	RESEND_REFRESH_TIME: 60,
+	MESSAGES_LIMIT: 50,
```
Comment (aminabhay) [inline src/infrastructure/constants.ts]: Remove unwanted.

File: src/infrastructure/navigation/AppNavigation.tsx [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+			{/* If Android then put a Custom Status Bar */}
+			{CONSTANTS.IS_ANDROID && (
+				<StatusBar
+					barStyle="light-content"
+					backgroundColor="transparent"
+					translucent
+				/>
+			)}
+
+			{/* If user is authenticated then show AuthStack else show HomeScreen */}
```
Comment (aminabhay) [inline src/infrastructure/navigation/AppNavigation.tsx]: capital U, A

In comments do not use camelCase, PascalCase words. Use normal english like Home Screen

show the Home Screen

File: src/services/api/users.api-service.ts [hunk 15]
```diff
@@ -0,0 +1,206 @@
...
+		const response = await axios.request<ApiResponseData<UserData>>(config);
+		return response.data;
+	} catch (error: any) {
+		return error.response;
+	}
+};
+
+/** Logs the User out of the App  */
+export const logoutRequest = async (
+	userId: string | null
```
Comment (aminabhay) [inline src/services/api/users.api-service.ts]: userId null kaise ho sakta hai

File: src/types/account.d.ts [hunk 16]
```diff
@@ -0,0 +1,16 @@
+type LoginApiData = {
+	user: UserCoreData;
+	token: string;
+};
+type LoginInputData = {
+	email_input: string;
+	password_input: string;
+};
+
+type LoginFormErrors = {
```
Comment (aminabhay) [inline src/types/account.d.ts]: end with Data

File: src/types/users.d.ts [hunk 17]
```diff
@@ -0,0 +1,32 @@
+interface Device {
```
Comment (aminabhay) [inline src/types/users.d.ts]: type

Comment (aminabhay) [reply src/types/users.d.ts]: DeviceData

File: src/types/users.d.ts [hunk 18]
```diff
@@ -0,0 +1,32 @@
...
+	email?: string;
+	phone_number?: string;
+	first_name?: string;
+	last_name?: string;
+	profile_photo?: string; // Relative path
+	password?: string;
+	username?: string;
+};
+
+interface UserCoreData {
```
Comment (aminabhay) [inline src/types/users.d.ts]: type

File: src/utils/validation.util.ts [hunk 19]
```diff
@@ -0,0 +1,6 @@
+/** Email Validation function */
+export const validateEmail = (email: string): boolean => {
+	// Regular expression to validate email format
+	const emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
+	return emailRegex.test(email);
+};
```
Comment (aminabhay) [inline src/utils/validation.util.ts:6]: Phone Number ka bhi validation Function bana do

**Review by aminabhay — CHANGES_REQUESTED** [review]

File: src/contexts/authentication.context.tsx [hunk 20]
```diff
@@ -0,0 +1,218 @@
...
+		flushLocalStorage();
+		setIsAuthLoading(false);
+	};
+
+	/** Logout the User */
+	const logout = async (): Promise<void> => {
+		// Start changing Auth State
+		setIsAuthLoading(true);
+
+		// Get the userId from Local storage
```
Comment (aminabhay) [inline src/contexts/authentication.context.tsx]: User ID

File: src/enums/analytics.ts [hunk 21]
```diff
@@ -0,0 +1,4 @@
+export enum AnalyticsEvents {
+	ACCOUNT_DELETED = "account_deleted",
+	USER_SIGN_UP = "user_sign_up",
```
Comment (aminabhay) [inline src/enums/analytics.ts]: Add to this

USER_LOGIN
USER_LOGOUT

File: src/features/account/screens/Login.screen.tsx [hunk 22]
```diff
@@ -0,0 +1,170 @@
...
+					form_error: "Looks like we faced some network issues. Please try again.",
+				}));
+			}
+		}
+	};
+
+	// Use Effect and Focus Effect
+	useEffect(() => {
+		// Log Event in analytics when Login Screen is opened
+		logPageViewEvent("Login");
```
Comment (aminabhay) [inline src/features/account/screens/Login.screen.tsx]: Make a new enum in analytics.enum.ts

PageNames
LOGIN = "Login"

And use it from here

File: src/infrastructure/constants.ts [hunk 23]
```diff
@@ -0,0 +1,18 @@
...
+	OS: Platform.OS,
+	IS_ANDROID: Platform.OS === "android",
+	IS_IOS: Platform.OS === "ios",
+	VERSION: Platform.Version,
+};
+
+// API URL
+export const API_URL = "https://api.pixoloproductions.com/";
+
+// Project ID
```
Comment (aminabhay) [inline src/infrastructure/constants.ts]: write in bracket (Expo Project ID)

File: src/services/analytics.service.ts [hunk 24]
```diff
@@ -0,0 +1,37 @@
...
+	// Get the userId from Local storage
+	const userId = (await getDataFromLocalStorage(LocalStorageKeys.USER))._id;
+	// Constants.appOwnership !== "expo" && analytics().logEvent("screen_view", {
+	// 	user: userId,
+	// 	screen: pageName,
+	// 	purpose: "Viewed the screen",
+	// });
+};
+
+/**  Will log the analytics events */
```
Comment (aminabhay) [inline src/services/analytics.service.ts]: capital A

File: src/services/analytics.service.ts [hunk 25]
```diff
@@ -0,0 +1,37 @@
...
+
+// SERVICES //
+import { getDataFromLocalStorage } from "./local-storage.service";
+
+// UTILS //
+import Constants from "expo-constants";
+
+/** Will log the analytics screen view event */
+export const logPageViewEvent = async (pageName: string) => {
+	// Get the userId from Local storage
```
Comment (aminabhay) [inline src/services/analytics.service.ts]: User ID

File: src/services/analytics.service.ts [hunk 26]
```diff
@@ -0,0 +1,37 @@
...
+// ENUMS //
+import { LocalStorageKeys } from "../enums/local-storage";
+
+// SERVICES //
+import { getDataFromLocalStorage } from "./local-storage.service";
+
+// UTILS //
+import Constants from "expo-constants";
+
+/** Will log the analytics screen view event */
```
Comment (aminabhay) [inline src/services/analytics.service.ts]: Capital A 
Screen View

**Review by aminabhay — CHANGES_REQUESTED** [review]

**Review by aminabhay — APPROVED** [review]

**Pull request merged**